x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -110,7 +110,7 @@ public TransportEsqlQueryAction(
         this.threadPool = threadPool;
         this.planExecutor = planExecutor;
         this.clusterService = clusterService;
-        this.requestExecutor = threadPool.executor(ThreadPool.Names.SEARCH);
+        this.requestExecutor = threadPool.executor(ThreadPool.Names.SEARCH_COORDINATION);
```

Comment:
Field caps is executed on `SEARCH_COORDINATION`.
This sound like query metadata processing should go to `SEARCH_COORDINATION` and actual searching/aggregation to `SEARCH`/`SYSTEM_READ` to me.